### DISCUSSION TITLE
How to set the progressIndicatorTheme in ThemeData

### DISCUSSION THREAD

**Discussion by freemansoft** [rydmike/flex_color_scheme#177] [Q&A] [answered]
I need to style our linear progress indicator background color.  

I create our theme using what I copied from the code popup the design tool

```
ThemeData appThemeLight = FlexThemeData.light(...
ThemeData appThemeDark = FlexThemeData.dark(...
```

The progressIndicatorTheme is empty/null.  The standard ThemeData class lets me do this

```
ThemeData foo = ThemeData(  progressIndicatorTheme: ProgressIndicatorThemeData(
      color: , 
      circularTrackColor: , 
      linearTrackColor: 
      refreshBackgroundColor:
  ));
```

I don't see the same progressIndicatorTheme on any of the flex_color_scheme classes.

**Comment by rydmike** [comment, marked as answer]
Hi @freemansoft,

Thanks for your question.

FlexColorScheme does not claim to contain all features that ThemeData has, only to make many of its more complex topics simpler 🙂

It does however return vanilla `ThemeData` so you can modify the returned object and add features to it that it lacks with normal `ThemeData` methods like `copyWith` and modify or add anything `ThemeData` offers with it, this in addition to what FlexSeedScheme has already done. For example

```dart
FlexThemeData.light(...).copyWith(progressIndicatorTheme: ProgressIndicatorThemeData(
      color: , 
      circularTrackColor: , 
      linearTrackColor: 
      refreshBackgroundColor:
  )));
```

You might want to look into other similar QAs here that shows more in depth examples and explanations on this, and also on how to access the ColorScheme it will create, as you might need its colors in your component theme that you add with `copyWith`.

Also please note that if you want to modify component themes it has already made, you need to operate the `copyWith` on it and not on an empty starting point. In the case of the 
`ProgressIndicatorThemeData` the situation is simpler though, as FlexColorScheme currently does not do anything to it.

I would link to the examples showing more info, but I am typing this answer on a phone, so it is a bit tricky. I can do so later if you need further assistance.

-Mike

**Comment by freemansoft** [comment]
Thanks.  That was what I needed to know.  I also appreciate the project scope information.

**Reply by rydmike** [reply]
Cool and you are welcome! 

Here are some links from another Q&As related to this topic https://github.com/rydmike/flex_color_scheme/discussions/116 and some explanations in the DOCs FAQ section
https://docs.flexcolorscheme.com/faq#can-i-use-the-colorscheme-flexcolorscheme-creates-in-custom-component-themes also contain info on this topic, as does this FAQ https://docs.flexcolorscheme.com/faq#how-do-i-modify-the-component-themes-with-custom-features, that discuss the steps when you want to override a property in a component theme and keep everything else it as it was, which we do not have to do in this particular `ProgressIndicatorThemeData ` component theme case.

Just adding them here in case someone else have similar question. 

Also, I am going to keep the question as "Answered" but open for discussion. I just do that since I have noticed that it makes it easier to find them in **Discussions** then, at least still with current relatively low volume.

**Reply by rydmike** [reply]
By the way, I just filed an issue about some lacking theming features in Flutter. 

Here: https://github.com/flutter/flutter/issues/131690

I think once those are fixed/added in Flutter I will add easy theming props for Progress Indicators in FCS and Themes Playground too.
